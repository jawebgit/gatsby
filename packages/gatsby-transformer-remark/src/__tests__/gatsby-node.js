const Promise = require(`bluebird`)

const { onNodeCreate } = require(`../gatsby-node`)

describe(`Process markdown content correctly`, () => {
  const node = {
    id: `whatever`,
    children: [],
    internal: {
      contentDigest: `whatever`,
      mediaType: `text/x-markdown`,
    },
  }

  // Make some fake functions its expecting.
  const loadNodeContent = node => Promise.resolve(node.content)

  it(`Correctly creates a new MarkdownRemark node`, async () => {
    const content = `---
title: "my little pony"
date: "12/12/2014"
---

Where oh where is my little pony?
    `
    node.content = content

    const createNode = jest.fn()
    const addNodeToParent = jest.fn()
    const boundActionCreators = { createNode, addNodeToParent }

    await onNodeCreate({
      node,
      loadNodeContent,
      boundActionCreators,
    }).then(() => {
      expect(createNode.mock.calls).toMatchSnapshot()
      expect(addNodeToParent.mock.calls).toMatchSnapshot()
      expect(createNode).toHaveBeenCalledTimes(1)
      expect(addNodeToParent).toHaveBeenCalledTimes(1)
    })
  })
})
