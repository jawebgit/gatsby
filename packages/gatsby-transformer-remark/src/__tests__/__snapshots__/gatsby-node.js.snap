// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Process markdown content correctly Correctly creates a new MarkdownRemark node 1`] = `
Array [
  Array [
    Object {
      "children": Array [],
      "content": "
Where oh where is my little pony?
    ",
      "contentDigest": "54cfa2ad99756f2fa232cac585280a37",
      "frontmatter": Object {
        "date": "12/12/2014",
        "parent": "whatever",
        "title": "my little pony",
      },
      "id": "whatever >>> MarkdownRemark",
      "mediaType": "text/x-markdown",
      "parent": "whatever",
      "type": "MarkdownRemark",
    },
  ],
]
`;

exports[`Process markdown content correctly Correctly creates a new MarkdownRemark node 2`] = `
Array [
  Array [
    Object {
      "children": Array [
        "whatever >>> MarkdownRemark",
      ],
      "content": "---
title: \\"my little pony\\"
date: \\"12/12/2014\\"
---

Where oh where is my little pony?
    ",
      "contentDigest": "whatever",
      "id": "whatever",
      "mediaType": "text/x-markdown",
    },
  ],
]
`;
